onnxruntime/core/session/custom_ops.cc

```diff
@@ -119,8 +119,19 @@ common::Status CreateCustomRegistry(const std::vector<OrtCustomOpDomain*>& op_do
       KernelDefBuilder def_builder;
       def_builder.SetName(op->GetName(op))
           .SetDomain(domain->domain_)
-          .SinceVersion(1)
-          .Provider(onnxruntime::kCpuExecutionProvider);
+          .SinceVersion(1);
+      if (&op->GetExecutionProviderType) {
+        auto provider = op->GetExecutionProviderType(op);
+        if (provider == ORT_EXECUTION_PROVIDER_CPU)
+          def_builder.Provider(onnxruntime::kCpuExecutionProvider);
+        else if (provider == ORT_EXECUTION_PROVIDER_CUDA)
+          def_builder.Provider(onnxruntime::kCudaExecutionProvider);
+        else
+          return Status(common::ONNXRUNTIME, common::FAIL, "Unsupported ExecutionProvider type");
```

Comment:
please also add all other supports as defined in "enum" above.

Reply:
every provider is supported now.